### PR TITLE
Added Migration for PGSQL and Collection for Typrsense

### PR DESCRIPTION
**Pull request by WahidinAji**
Closes # 

## 📑 Deskripsi
image

I've added the migration of the Yugabyte / PGSQL schema and Collections of Typesense.
The documentation isn't ready yet, It will update in the next PR (^-^)

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: hack/provider/shcema.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
+package hack_provider
```
Comment (aldy505) [inline hack/provider/shcema.go]: typo filename. harusnya schema.

File: hack/provider/shcema.go [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+		return &hack_stub.HackServiceError{StatusCode: 500, Error: fmt.Errorf("message err failed to connect to database:  %w", err)}
+	}
+	defer func() {
+		err := db.Close()
+		if err != nil && !errors.Is(err, sql.ErrConnDone) {
+			log.Printf("failed to close database connection: %V", err)
+		}
+	}()
+
+	tx, err := db.BeginTx(ctx, &sql.TxOptions{Isolation: sql.LevelReadCommitted})
```
Comment (aldy505) [inline hack/provider/shcema.go]: ini harusnya serializable

File: hack/provider/shcema.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+	if err != nil {
+		return &hack_stub.HackServiceError{StatusCode: 500, Error: fmt.Errorf("failed to begin transaction: %w", err)}
+	}
+	defer func() {
+		if errRollback := tx.Rollback(); errRollback != nil {
+			log.Printf("message err rollback create new hack post: %v", err.Error())
```
Comment (aldy505) [inline hack/provider/shcema.go]: message errornya ga jelas. jangan pake %v kalo udah tau tipe datanya

File: hack/provider/shcema.go [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+	log.Printf("Migration successfully!")
+	return nil
+
+}
+
+func CreateCollections(ctx context.Context, client *typesense.Client) *hack_stub.HackServiceError {
+	// authors collection
+	_, err := client.Collection("authors").Retrieve()
+	if err != nil {
+		log.Printf("authors collection does not exist: %v\n", err.Error())
```
Comment (aldy505) [inline hack/provider/shcema.go]: harus pastiin kalo errornya karena does not exist. terus jangan pake %v

File: hack/provider/shcema.go [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+		_, err = client.Collections().Create(authorColletions)
+		if err != nil {
+			log.Fatalf("created authors collection error: %v", err.Error())
+		}
+	}
+
+	//hacks collection
+	_, err = client.Collection("hacks").Retrieve()
+	if err != nil {
+		log.Printf("hacks collection does not exist: %v\n", err.Error())
```
Comment (aldy505) [inline hack/provider/shcema.go]: harus pastiin kalo error karena does not exist

File: hack/provider/shcema.go [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+		_, err = client.Collections().Create(commentColletion)
+		if err != nil {
+			log.Fatalf("created hacks collection error: %v", err)
+		}
+	}
+
+	//comments collection
+	_, err = client.Collection("comments").Retrieve()
+	if err != nil {
+		log.Printf("comments collection does not exist: %v\n", err.Error())
```
Comment (aldy505) [inline hack/provider/shcema.go]: harus pastiin karena does not exist

File: hack/provider/shcema.go [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+	"errors"
+	"fmt"
+	hack_stub "kodiiing/hack/stub"
+	"log"
+
+	"github.com/typesense/typesense-go/typesense"
+	"github.com/typesense/typesense-go/typesense/api"
+)
+
+func MigrateYugabyte(ctx context.Context, d *sql.DB) *hack_stub.HackServiceError {
```
Comment (aldy505) [inline hack/provider/shcema.go]:
```suggestion
func MigrateHackSQL(ctx context.Context, d *sql.DB) *hack_stub.HackServiceError {
```

File: main.go [hunk 8]
```diff
@@ -80,6 +81,19 @@ func main() {
 		}
 	}(memory)
 
+	//context
+	ctx := context.Background()
+	//schema migration (YugaByte/PGSQL)
+	errMigrateSchema := hack_provider.MigrateYugabyte(ctx, db)
```
Comment (aldy505) [inline main.go]:
```suggestion
	errMigrateSchema := hack_provider.MigrateHackSQL(ctx, db)
```

File: main.go [hunk 9]
```diff
@@ -80,6 +81,19 @@ func main() {
 		}
 	}(memory)
 
+	//context
+	ctx := context.Background()
```
Comment (aldy505) [inline main.go]: jangan pake context.Background, bikin timeout. atau kalo nggak tau mau apa, pake context.TODO

**Review by aldy505 — CHANGES_REQUESTED** [review]

File: hack/provider/shcema.go (hunk 4 above)
Comment (WahidinAji) [reply hack/provider/shcema.go]: itu `Retieve()` udah mastiin om. jadi kalau ada dia nampilin detail dari collectionnya kan, nah kalau ga ada balikannya `nil` . makanya disitu pake underscore `_` soale kalau ada ya udah tinggalin. bgitu

**Review by aldy505 — APPROVED** [review]

**Pull request merged**